src/native/eventpipe/ds-eventpipe-protocol.c

```diff
@@ -195,89 +292,299 @@ eventpipe_collect_tracing_command_try_parse_stackwalk_requested (
+	ep_raise_error_if_nok (*tracepoint_config != NULL);
+
+	ep_raise_error_if_nok (ds_ipc_message_try_parse_string_utf16_t_string_utf8_t_alloc (buffer, buffer_len, &(*tracepoint_config)->default_tracepoint_name));
+
+	ep_raise_error_if_nok (ds_ipc_message_try_parse_uint32_t (buffer, buffer_len, &(*tracepoint_config)->non_default_tracepoints_length));
+
+	ep_raise_error_if_nok (eventpipe_collect_tracing_command_try_parse_tracepoint_sets (buffer, buffer_len, (*tracepoint_config)->non_default_tracepoints_length, &(*tracepoint_config)->non_default_tracepoints));
+
+	ep_raise_error_if_nok ((*tracepoint_config)->default_tracepoint_name != NULL || (*tracepoint_config)->non_default_tracepoints_length > 0);
+
+	result = true;
+
+ep_on_exit:
+	return result;
+
+ep_on_error:
+	eventpipe_collect_tracing_command_free_tracepoint_config (*tracepoint_config);
+	*tracepoint_config = NULL;
+	ep_exit_error_handler ();
+}
+
+/*
+ *  eventpipe_collect_tracing_command_try_parse_provider_config
+ *
+ *  Deserializes a single EventPipeProviderConfiguration from the IPC Stream
+ *
+ *  Dynamically allocates memory for the EventPipeProviderConfiguration fields and passes ownership to the caller.
+ */
+static
+bool
+eventpipe_collect_tracing_command_try_parse_provider_config (
+	uint8_t **buffer,
+	uint32_t *buffer_len,
+	EventPipeProviderOptionalFieldFlags optional_field_flags,
+	EventPipeProviderConfiguration *provider_config)
+{
+	EP_ASSERT (buffer != NULL);
+	EP_ASSERT (buffer_len != NULL);
+	EP_ASSERT (provider_config != NULL);
+
+	bool result = false;
+
+	provider_config->keywords = 0;
+	provider_config->logging_level = (EventPipeEventLevel)0;
+	provider_config->provider_name = NULL;
+	provider_config->filter_data = NULL;
+	provider_config->event_filter = NULL;
+	provider_config->tracepoint_config = NULL;
+
+	ep_raise_error_if_nok (ds_ipc_message_try_parse_uint64_t (buffer, buffer_len, &provider_config->keywords));
+
+	ep_raise_error_if_nok (eventpipe_collect_tracing_command_try_parse_logging_level (buffer, buffer_len, &provider_config->logging_level));
+
+	ep_raise_error_if_nok (ds_ipc_message_try_parse_string_utf16_t_string_utf8_t_alloc (buffer, buffer_len, &provider_config->provider_name));
+	ep_raise_error_if_nok (!ep_rt_utf8_string_is_null_or_empty (provider_config->provider_name));
+
+	ep_raise_error_if_nok (ds_ipc_message_try_parse_string_utf16_t_string_utf8_t_alloc (buffer, buffer_len, &provider_config->filter_data));
+
+	if ((optional_field_flags & EP_PROVIDER_OPTFIELD_EVENT_FILTER) != 0)
+		ep_raise_error_if_nok (eventpipe_collect_tracing_command_try_parse_event_filter (buffer, buffer_len, &provider_config->event_filter));
+
+	if ((optional_field_flags & EP_PROVIDER_OPTFIELD_TRACEPOINT_CONFIG) != 0)
+		ep_raise_error_if_nok (eventpipe_collect_tracing_command_try_parse_tracepoint_config (buffer, buffer_len, &provider_config->tracepoint_config));
+
+	result = true;
+
+ep_on_exit:
+	return result;
+
+ep_on_error:
+	ep_exit_error_handler ();
```

Comment:
To be consistent (and already pointed out), we should free the provider_config on failure and not put that responsibility on caller.

Reply:
Maybe we should handle that scenario directly in that code, like:

```
ep_raise_error_if_nok (eventpipe_collect_tracing_command_try_parse_provider_config (
			buffer,
			buffer_len,
			optional_field_flags,
			&provider_config));

if (!dn_vector_push_back (*result, provider_config))
{
    ep_config_provider_fini(&provider_config);
}

ep_config_provider_clear(&provider_config);
```

and then we could drop the ep_provider_config_fini from the error scenario and replace it with freeing up all items currently in the vector. To make it even more clear you can move the:

`EventPipeProviderConfiguration provider_config;`

into the for loop, then it won't have scope outside unless transferred over to vector.

Reply:
updated to be 
```C
	for (uint32_t i = 0; i < count_configs; ++i) {
		EventPipeProviderConfiguration provider_config;
		ep_raise_error_if_nok (eventpipe_collect_tracing_command_try_parse_provider_config (
			buffer,
			buffer_len,
			optional_field_flags,
			&provider_config));

		if (!dn_vector_push_back (*result, provider_config))
			ep_provider_config_fini (&provider_config);
	}
```
So the struct is locally scoped, and allocated resources are freed in case it fails to be copied into the vector